modules/3d/test/test_mst.cpp

```diff
@@ -0,0 +1,267 @@
+        },
+        {
+            {0, 1, 0.0}, {1, 2, 1e-9}
+        }
+    ),
+
+    MSTParamType(MST_KRUSKAL, 3,
+        {
+            {0, 1, 0.0}, {1, 2, 1e-9}, {0, 2, 1.0}
+        },
+        {
+            {0, 1, 0.0}, {1, 2, 1e-9}
+        }
+    ),
+
+    // Duplicate edges (picks the one with the smallest weight)
+    MSTParamType(MST_PRIM, 3,
+        {
+            {0, 1, 3.0}, {0, 1, 1.0}, {1, 2, 2.0}
+        },
+        {
+            {0, 1, 1.0}, {1, 2, 2.0}
+        }
+    ),
+
+    MSTParamType(MST_KRUSKAL, 3,
+        {
+            {0, 1, 3.0}, {0, 1, 1.0}, {1, 2, 2.0}
+        },
+        {
+            {0, 1, 1.0}, {1, 2, 2.0}
+        }
+    ),
+
+    // Negative weights
+    MSTParamType(MST_PRIM, 3,
+        {
+            {0, 1, -1.0}, {1, 2, -2.0}, {0, 2, -3.0}
+        },
+        {
+            {1, 2, -2.0}, {0, 2, -3.0}
+        }
+    ),
+
+    MSTParamType(MST_KRUSKAL, 3,
+        {
+            {0, 1, -1.0}, {1, 2, -2.0}, {0, 2, -3.0}
+        },
+        {
+            {0, 2, -3.0}, {1, 2, -2.0}
+        }
+    ),
+};
+
+inline static std::string MST_name_printer(const testing::TestParamInfo<MST::ParamType>& info)
+{
+    std::ostringstream os;
+    const auto& algorithm = get<0>(info.param);
+    const auto& numNodes = get<1>(info.param);
+    const auto& edges = get<2>(info.param);
+    const auto& expectedEdges = get<3>(info.param);
+
+    os << "TestCase_" << info.index << "_";
+    switch (algorithm)
+    {
+    case MST_PRIM: os << "Prim"; break;
+    case MST_KRUSKAL: os << "Kruskal"; break;
+    default: os << "Unknown algorithm"; break;
+    }
+    os << "_Nodes_" << numNodes;
+    os << "_Edges_" << edges.size();
+    os << "_ExpectedEdges_" << expectedEdges.size();
+
+    return os.str();
+}
+
+INSTANTIATE_TEST_CASE_P(/**/, MST, testing::ValuesIn(mst_graphs), MST_name_printer);
+
+TEST(MSTstress, LargeGraph)
+{
+    const int numNodes = 100000;
+
+    std::vector<MSTEdge> edges;
+
+    for (int i = 0; i < numNodes - 1; ++i)
+        edges.push_back({i, i + 1, static_cast<double>(i + 1)});
+
+    // Add extra edges for complexity
+    for (int i = 0; i < numNodes - 10; i += 10)
+        edges.push_back({i, i + 10,  static_cast<double>(i)});
+    for (int i = 0; i + 20 < numNodes; i += 5)
+        edges.push_back({i, i + 20, static_cast<double>(i + 1)});
+    for (int i = 0; i + 30 < numNodes; i += 3)
+        edges.push_back({i, i + 30, static_cast<double>(i % 50 + 1)});
+    for (int i = 50; i < numNodes; i += 10)
+        edges.push_back({i, i - 25, static_cast<double>(i % 100 + 2)});
+
+    std::vector<MSTEdge> primMST, kruskalMST;
+    bool resultPrim = buildMST(numNodes, edges, primMST, MST_PRIM, 0);
+    bool resultKruskal = buildMST(numNodes, edges, kruskalMST, MST_KRUSKAL, 0);
+
+    EXPECT_TRUE(resultPrim);
+    EXPECT_TRUE(resultKruskal);
+    EXPECT_EQ(primMST.size(), static_cast<size_t>(numNodes - 1));
+    EXPECT_EQ(kruskalMST.size(), static_cast<size_t>(numNodes - 1));
+}
```

Comment:
Please add a couple of tests with negative scenarios:
- split graph of 2-3 sub-graphs
- one isolated node in the graph